### PR TITLE
Polysemy: Separate more Cassandra effects from Spar

### PR DESCRIPTION
**Pull request by isovector**
This PR rips out a bunch more effects from Spar for polysemization and eventual unit-testing support: 

* `ScimExternalIdStore`
* `ScimUserTimesStore`
* `ScimTokenStore`
* `DefaultSsoCode`

Also, it renames `SAMLUser` to `SAMLUserStore`.

To be rebased onto `develop` after #1787 lands.

## Checklist

 - [x] The **PR Title** explains the impact of the change.
 - [x] The **PR description** provides context as to why the change should occur and what the code contributes to that effect. This could also be a link to a JIRA ticket or a Github issue, if there is one.
 - [x] **changelog.d** contains the following bits of information:
   - [x] A file with the changelog entry in one or more suitable sub-sections. The sub-sections are marked by directories inside `changelog.d`.

### CODE REVIEW

**Review by fisx — CHANGES_REQUESTED** [review]
pls rebase or merge

File: services/spar/src/Spar/App.hs [hunk 1]
```diff
@@ -415,15 +425,27 @@ bindUser buid userref = do
...
+            embedToFinal @IO $
+              interpretClientToIO (sparCtxCas ctx) $
+                samlUserStoreToCassandra @Cas.Client $
+                  idPToCassandra @Cas.Client $
+                    defaultSsoCodeToCassandra $
+                      scimTokenStoreToCassandra $
+                        scimUserTimesStoreToCassandra $
+                          scimExternalIdStoreToCassandra $
+                            runExceptT $
+                              runReaderT action ctx
```
Comment (fisx) [inline services/spar/src/Spar/App.hs:448]: pretty!  :)

File: services/spar/src/Spar/Scim/User.hs [hunk 2]
```diff
@@ -798,7 +804,13 @@ scimFindUserByHandle mIdpConfig stiTeam hndl = do
...
 -- Note the user won't get an entry in `spar.user`.  That will only happen on their first
 -- successful authentication with their SAML credentials.
-scimFindUserByEmail :: forall r. Member SAMLUser r => Maybe IdP -> TeamId -> Text -> MaybeT (Scim.ScimHandler (Spar r)) (Scim.StoredUser ST.SparTag)
+scimFindUserByEmail ::
+  forall r.
+  Members '[ScimExternalIdStore, ScimUserTimesStore, SAMLUserStore] r =>
+  Maybe IdP ->
+  TeamId ->
+  Text ->
+  MaybeT (Scim.ScimHandler (Spar r)) (Scim.StoredUser ST.SparTag)
```
Comment (fisx) [inline services/spar/src/Spar/Scim/User.hs:813]: I like that you made the type signature multi-line here (and in some other places).

File: services/spar/src/Spar/Sem/DefaultSsoCode/Cassandra.hs [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  forall m r a.
+  (MonadClient m, Member (Embed m) r) =>
+  Sem (DefaultSsoCode ': r) a ->
+  Sem r a
+defaultSsoCodeToCassandra =
+  interpret $
+    embed @m . \case
+      Get -> Data.getDefaultSsoCode
+      Store ip -> Data.storeDefaultSsoCode ip
+      Delete -> Data.deleteDefaultSsoCode
```
Comment (fisx) [inline services/spar/src/Spar/Sem/DefaultSsoCode/Cassandra.hs:21]: I suppose at some point we can resolve `Spar.Data` into these intepreter modules, what do you think?

**Review by fisx — APPROVED** [review]
all good to go!

File: services/spar/src/Spar/Sem/DefaultSsoCode/Cassandra.hs (hunk 3 above)
Comment (isovector) [reply services/spar/src/Spar/Sem/DefaultSsoCode/Cassandra.hs:21]: That sounds good to me!

**Pull request merged**